### PR TITLE
Transition to standard smart pointers, part 6

### PR DESCRIPTION
**Pull request by taketwo**
Partially addresses #2792.

### CODE REVIEW

File: apps/src/organized_segmentation_demo.cpp [hunk 1]
```diff
@@ -54,9 +54,10 @@ displayEuclideanClusters (const pcl::PointCloud<PointT>::CloudVectorType &cluste
   for (size_t i = 0; i < clusters.size (); i++)
   {
     sprintf (name, "cluster_%d" , int (i));
-    pcl::visualization::PointCloudColorHandlerCustom<PointT> color0(boost::make_shared<pcl::PointCloud<PointT> >(clusters[i]),red[i%6],grn[i%6],blu[i%6]);
-    if (!viewer->updatePointCloud (boost::make_shared<pcl::PointCloud<PointT> >(clusters[i]),color0,name))
-      viewer->addPointCloud (boost::make_shared<pcl::PointCloud<PointT> >(clusters[i]),color0,name);
+    pcl::PointCloud<PointT>::Ptr cluster_cloud (new pcl::PointCloud<PointT> (clusters[i]));
+    pcl::visualization::PointCloudColorHandlerCustom<PointT> color0(cluster_cloud,red[i%6],grn[i%6],blu[i%6]);
+    if (!viewer->updatePointCloud (cluster_cloud,color0,name))
+      viewer->addPointCloud (cluster_cloud,color0,name);
```
Comment (SergioRAgostinho) [inline apps/src/organized_segmentation_demo.cpp]: The prior behavior was to generate shared_pointers and invoke separate copy ctors i.e., every time make_shared is invoked, a copy is being created. 

With the new behavior, the copy ctor is invoked only once and then you simply pass the same shared pointer around. This is not a `const` shared pointer so I'm slightly concerned these methods might have the freedom to modify the contents. Hopefully, if the interface was done right, we can quickly test this by setting
```cpp
pcl::PointCloud<PointT>::ConstPtr cluster_cloud (new pcl::PointCloud<PointT> (clusters[i]));
```
and see if it still compiles.

File: apps/src/openni_fast_mesh.cpp [hunk 2]
```diff
@@ -89,18 +89,13 @@ class OpenNIFastMesh
...
       // Lock and copy
       {
         boost::mutex::scoped_lock lock (mtx_);
-        //boost::unique_lock<boost::shared_mutex> lock (mtx_);
-
-//        if (!vertices_)
-//          vertices_.reset (new std::vector<pcl::Vertices>);
-        //vertices_.reset (new std::vector<pcl::Vertices> (*temp_verts));
-        vertices_= temp_verts;
+        std::swap(vertices_, temp_verts);
```
Comment (SergioRAgostinho) [inline apps/src/openni_fast_mesh.cpp]: shouldn't this be a 
```cpp
vertices_ = std::move (temp_verts);
```
instead?

File: apps/src/openni_fast_mesh.cpp [hunk 3]
```diff
@@ -129,12 +124,12 @@ class OpenNIFastMesh
         }
 
         temp_cloud = cloud_;
-        temp_verts = vertices_;
+        std::swap(temp_verts, vertices_);
```
Comment (SergioRAgostinho) [inline apps/src/openni_fast_mesh.cpp]: Same comment as before.

File: apps/src/organized_segmentation_demo.cpp (hunk 1 above)
Comment (taketwo) [reply apps/src/organized_segmentation_demo.cpp]: Both `updatePointCloud` and `addPointCloud` take `ConstPtr`, so they don't modify anything. And yeah, it still compiles with your suggestion. Do you want me to commit this change?

File: apps/src/openni_fast_mesh.cpp (hunk 2 above)
Comment (taketwo) [reply apps/src/openni_fast_mesh.cpp]: Well yes, this would be more concise.

File: apps/src/organized_segmentation_demo.cpp (hunk 1 above)
Comment (taketwo) [reply apps/src/organized_segmentation_demo.cpp]: Committed anyway

**Pull request merged**